ui/pages/confirmations/components/confirm/info/shared/transaction-details.tsx

```diff
@@ -0,0 +1,78 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { isValidAddress } from 'ethereumjs-util';
+import React from 'react';
+import { useSelector } from 'react-redux';
+import {
+  ConfirmInfoRow,
+  ConfirmInfoRowAddress,
+  ConfirmInfoRowText,
+  ConfirmInfoRowUrl,
+} from '../../../../../../components/app/confirm/info/row';
+import { useI18nContext } from '../../../../../../hooks/useI18nContext';
+import { currentConfirmationSelector } from '../../../../selectors';
+import { useKnownMethodDataInTransaction } from '../hooks/known-method-data-in-transaction';
+
+const OriginRow = () => {
+  const t = useI18nContext();
+
+  const currentConfirmation = useSelector(
```

Comment:
These sub-components are great!

Minor, and personal preference, but they could have their own properties such as `currentConfirmation` so we don't have to duplicate this selector call multiple times, but I appreciate this one is only a line.

Reply:
I'd rather keep this as is for now if that's ok, we can create that prop later if duplication does get out of hand

Reply:
Of course, not a priority and approved the PR, just highlighting the possibility.